lib/client.js

```diff
@@ -219,80 +225,87 @@ exports.createClient = function(reference) {
-              var res = err.response;
-              if (res) {
-                // Decide if we should retry
-                if (attempts < that._options.maxRetries &&
-                    500 <= res.status &&  // Check if it's a 5xx error
-                    res.status < 600) {
-                  debug("Error calling: %s now retrying, info: %j",
-                        entry.name, res.body);
-                  return retryRequest();
-                }
-                // If not retrying, construct error object and reject
-                debug("Error calling: %s NOT retrying!, info: %j",
-                      entry.name, res.body);
-                var message = "Unknown Server Error";
-                if (res.status === 401) {
-                  message = "Authentication Error";
-                }
-                if (res.status === 500) {
-                  message = "Internal Server Error";
-                }
-                err = new Error(res.body.message || message);
-                err.body = res.body;
-                err.statusCode = res.status;
-                return reject(err);
-              }
-
+      // Retry the request, after a delay depending on number of retries
+      var retryRequest = function() {
+        // Send request
+        var sendRequest = function() {
+          debug("Calling: %s, retry: %s", entry.name, attempts - 1);
+          // Make request and handle response or error
+          return makeRequest(
+            that,
+            entry.method,
+            url,
+            payload
+          ).end().then(function(res) {
+            // If request was successful, accept the result
+            debug("Success calling: %s, (%s retries)",
+                  entry.name, attempts - 1);
+            return res.body;
+          }, function(err) {
+            // If we got a response we read the error code from the response
+            var res = err.response;
+            if (res) {
               // Decide if we should retry
-              if (attempts < that._options.maxRetries) {
-                debug("Request error calling %s (retrying), err: %s, JSON: %s",
-                      entry.name, err, err);
+              if (attempts <= that._options.retries &&
+                  500 <= res.status &&  // Check if it's a 5xx error
+                  res.status < 600) {
+                debug("Error calling: %s now retrying, info: %j",
+                      entry.name, res.body);
                 return retryRequest();
               }
-              debug("Request error calling %s NOT retrying!, err: %s, JSON: %s",
+              // If not retrying, construct error object and reject
+              debug("Error calling: %s NOT retrying!, info: %j",
+                    entry.name, res.body);
+              var message = "Unknown Server Error";
+              if (res.status === 401) {
+                message = "Authentication Error";
+              }
+              if (res.status === 500) {
```

Comment:
This code was already there... I just refactored it a bit... But lack of 503 handling is explains why requests don't get retried!!! Ups!
I'll go add that now... We should retry 503.